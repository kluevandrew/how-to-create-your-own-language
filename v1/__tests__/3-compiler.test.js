const { tokenize } = require('../1-tokenizer');
const { parse } = require('../2-parser');
const { compile } = require('../3-compiler');

describe('compile', () => {
  it('should compile', () => {
    expect(compile(parse(tokenize('(print (a))')))).toMatchSnapshot();
  });
});
