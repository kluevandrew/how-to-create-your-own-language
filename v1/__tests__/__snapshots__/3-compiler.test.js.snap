// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compile should compile 1`] = `
Function {
  "opcodes": Array [
    Opcode {
      "type": "LOAD_FAST",
      "value": "print",
    },
    Opcode {
      "type": "LOAD_FAST",
      "value": "a",
    },
    Opcode {
      "type": "CALL_FUNCTION",
      "value": 0,
    },
    Opcode {
      "type": "CALL_FUNCTION",
      "value": 1,
    },
    Opcode {
      "type": "POP_TOP",
      "value": 0,
    },
  ],
}
`;
